builtin/credential/approle/path_role.go

```diff
@@ -2358,12 +2359,27 @@ func (b *backend) handleRoleSecretIDCommon(ctx context.Context, req *logical.Req
 		return nil, errwrap.Wrapf("failed to store secret_id: {{err}}", err)
 	}
 
-	return &logical.Response{
+	resp := &logical.Response{
 		Data: map[string]interface{}{
 			"secret_id":          secretID,
 			"secret_id_accessor": secretIDStorage.SecretIDAccessor,
 		},
-	}, nil
+	}
+
+	// Follow the behavior from registerSecretIDEntry and set the response TTL
+	// to the entry's TTL unless it's negative or greater than the system's max
+	// lease TTL.
```

Comment:
If `registerSecretIDEntry` is setting the TTL on the entry do we need to duplicate the logic here or will secretIDStorage.SecretIDTTL already reflect the calculated TTL?

Reply:
`registerSecretIDEntry` calculates the expiration time (no duration) based off the `SecretIDTTL`, but it doesn't modify it which is why we need to check and cap it outside of the function.

Reply:
I'll extract the logic to a helper func that can be called in both places.

Reply:
Done in 8b90861.